R/regression_model.R

```diff
@@ -1,11 +1,18 @@
-new_regression_model <- function(design, outcome, model_name) {
-  model <- list(design = design, outcome = outcome, name = model_name)
+new_regression_model <- function(design, outcome, model_name, SGD_solver, via_transp) {
+  n_obs <- nrow(design)
+  n_pred <- ncol(design)
+  if (SGD_solver & via_transp) {
```

Comment:
Note that `&` and `&&` have different meanings and, while `&` is fine here, using `&&` is a more standard practice for conditional statements.

Reply:
No longer relevant b/c I'm not passing `SGD_solver` anymore